[wip] Thockin hack on deads2k gates branch

deads2k opened the pull request:
opening this because I cannot seem to comment on diffs

danwinship committed:
Merge linux and windows kube-proxy metric registration together

Windows proxy metric registration was in a separate file, which had
led to some metrics (eg the new ProxyHealthzTotal and ProxyLivezTotal)
not being registered for Windows even though they were implemented by
platform-generic code.

(A few other metrics were neither registered on, nor implemented on
Windows, and that's probably a bug.)

Also, beyond linux-vs-windows, make it clearer which metrics are
specific to individual backends.

sttts committed:
pkg/controlplane: split up config into generic controlplane and kube-related part

Signed-off-by: Dr. Stefan Schimanski <stefan.schimanski@gmail.com>

wackxu committed:
implement QueueingHint in TaintToleration

liyuerich committed:
drop deprecated workqueue NewNamed package

Signed-off-by: liyuerich <yue.li@daocloud.io>

benluddy committed:
Deduplicate set expression values in metav1.LabelSelector fuzzer.

Internal versions of ScaleStatus types use metav1.LabelSelector to represent label selectors, while
external versions use the textual representation. During conversion to and from text, match
expressions are sorted by key, and values for set operations "in" and "notin" are sorted and
deduplicated. This loss of order and duplication is detected by roundtrip testing.

The existing fuzz function for metav1.LabelSelector sorts match expressions by key and sorts, but
does not deduplicate, set expression values. That function now also deduplicates set expression
values so that fuzzed metav1.LabelSelectors can faithfully roundtrip through the textual label
selector representation.

danwinship committed:
Move LocalTrafficDetector from pkg/proxy/util/iptables to pkg/proxy/util

Since it's used for nftables as well now.

pohly committed:
cloud providers: enhance context support

27a68aee3a4834 introduced context support for events. Creating an event
broadcaster with context makes tests more resilient against leaking goroutines
when that context gets canceled at the end of a test and enables per-test
output via ktesting.

While at it, all context.TODO calls get removed in files that were touched.

pohly committed:
sample controller: enhance context support

27a68aee3a4834 introduced context support for events. Creating an event
broadcaster with context makes tests more resilient against leaking goroutines
when that context gets canceled at the end of a test and enables per-test
output via ktesting.

The New method already had a context, therefore no API changes are needed.

pohly committed:
apiserver + controllers: enhance context support

27a68aee3a4834 introduced context support for events. Creating an event
broadcaster with context makes tests more resilient against leaking goroutines
when that context gets canceled at the end of a test and enables per-test
output via ktesting.

The context could get passed to the constructor. A cleaner solution is to
enhance context support for the apiserver and then pass the context into the
controller's run method. This ripples up the call stack to all places which
start an apiserver.

danwinship committed:
fix "Iptables" -> "IPTables" in metrics variable names

dims committed:
Remove gcp in-tree cloud provider and credential provider

Signed-off-by: Davanum Srinivas <davanum@gmail.com>

danwinship committed:
Simplify redundant LocalTrafficDetector implementations

All of the LocalTrafficDetector implementations were essentially
identical after construction time, so just reduce them to a single
implementation with multiple constructors.

Also, improve the comments.

danwinship committed:
Remove errors from LocalTrafficDetector constructors

The constructors only return an error if you pass them invalid data,
but we only ever pass them data which has already been validated,
making the error checking just annoying. Just make them return garbage
output if you give them garbage input.

aroradaman committed:
kube-proxy: refactor config validation unit test

Refactor the TestValidateKubeProxyConfiguration by adding a mutating
function that adjusts the configuration according to each test case,
thereby enhancing readability.

Signed-off-by: Daman Arora <aroradaman@gmail.com>

Commit:
feature: add name formats library to CEL

saschagrunert committed:
Add `cri-client` staging repository

Signed-off-by: Sascha Grunert <sgrunert@redhat.com>

danwinship committed:
Use t.Run and assert.Equal in KubeProxyConfiguration validation tests

danwinship committed:
Add names to all KubeProxyConfiguration validation test cases

danwinship committed:
Merge success and failure cases in KubeProxyConfiguration validation tests

saschagrunert committed:
Decouple `kubelet/cri/remote` package from `pkg/features`

Importing the `k8s.io/kubernetes/pkg/features` package in the remote
runtime implementation makes it harder to separate the functionalities
at some later point in time.

We now decouple them by checking if the feature is enabled directly in
the kubelet service creation path.

Signed-off-by: Sascha Grunert <sgrunert@redhat.com>

thockin committed:
DO NOT MERGE: hack on k8s.io/utils via staging

$ go list -json k8s.io/utils/ptr | jq .Dir
"/home/thockin/src/kubernetes/staging/src/k8s.io/utils/ptr"

claudiubelu committed:
unittests: Skip failing Windows tests

Some of the unit tests are currently failing on Windows.

Skip them for now, and remove the skips later, once the underlying issues
have been resolved.

pohly committed:
scheduler: move assume cache to utils, part 2

This is now used by both the volumebinding and dynamicresources plugin, so
promoting it to a common helper package is better.

In terms of functionality, nothing was changed. Documentation got
updated (warns about storing locally modified objects, clarifies what the Get
parameters are). Code coverage should be a bit better than before (tested with
and without indexer, exercises event handlers, more error paths).

Checking for specific errors can now be done via errors.Is.

toddtreece committed:
Add tracing to aggregator proxyHandler

deads2k commented on pkg/api/pod/util.go:
```diff
@@ -685,7 +686,8 @@ func dropDisabledFields(
 
-	if !utilfeature.DefaultFeatureGate.Enabled(features.RecursiveReadOnlyMounts) && !rroInUse(oldPodSpec) {
+	if !utilfeature.DefaultFeatureGate.Enabled(features.RecursiveReadOnlyMounts) &&
+		!feature.Enabled(features.RecursiveReadOnlyMounts3) && !rroInUse(oldPodSpec) {
```
I think in my refactor this became `!features.RecursiveReadOnlyMounts3.Enabled()` .  I guess the advantage in this refactor is that every featuregate has the env var assignment?  How attached are you to the `feature.Enabled` ?

deads2k commented on cmd/kube-controller-manager/app/storageversionmigrator.go:
```diff
@@ -45,8 +46,8 @@ func startSVMController(
-	if !utilfeature.DefaultFeatureGate.Enabled(features.StorageVersionMigrator) ||
-		!clientgofeaturegate.InformerResourceVersion2.Enabled() {
+	if !oldutilfeature.DefaultFeatureGate.Enabled(features.StorageVersionMigrator) &&
+		!feature.Enabled(clientgofeaturegate.InformerResourceVersion3) {
```
I think in my refactor this became `!features.RecursiveReadOnlyMounts3.Enabled()`. I guess the advantage in this refactor is that every featuregate has the env var assignment? How attached are you to the `feature.Enabled` ?

thockin replied:
I don't see why we wouldn't want ALL gates to have the same enablement options?

`feature.Enabled(` syntax was JUST to put the verb at the front of the LOC.  I find it easier to read as:

```
if !feature.Enabled(longpkgname.bigLongVariableName) {
```

than

```
if !longpkgname.bigLongVariableName.Enabled() {
```

That's REALLY the only argument :)

deads2k commented on staging/src/k8s.io/utils/feature/feature.go:
```diff
@@ -0,0 +1,263 @@
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+)
+
+//FIXME: unclear if we want to make the API in terms of "Gates" or "Features".
+//"Features" is so abstract...
```
I prefer the more explicit `FeatureGate` and `FeatureGateSet` (or `FeatureSet`)

thockin replied:
I was trying to follow the "don't stutter" approach.  `feature.Gate` vs `feature.FeatureGate` or heaven forbid `featuregate.FeatureGate`.

deads2k commented on staging/src/k8s.io/client-go/features/known_features.go:
```diff
@@ -68,6 +81,11 @@ func LibraryFeatureSet() featuregates.FeatureSet {
 }
 
+// FIXME: rename to FeatureGates when other definition is gone.
+func LibFeatureGates() *feature.GateSet {
```
I see an advantage to using an interface over a concrete type here so that in composition people can say, "David and Tim won't are too slow to merge my awesome way to set FeatureGates, so I'll provide a compatible implementation myself!" and anything accepting a GateSet will "just work".

thockin replied:
I see that as exceedingly unlikely, or at least not super compelling vs. the complexity.  Using interfaces has a few costs, IMO:

1) Harder to keep in brain
2) MUCH harder to back-track to "where did I get called from"
3) Requires everything be methods, meaning the types get more obtuse
4) Encourages defining more and more interfaces when a "dumb" struct would do
5) Encourages "SimpleFeatureGate" stuff which makes it more difficult again to track

Aside: I looked again at your branch and I honestly have no idea what a private method in a public interface means :)

I won't die on this hill, probably, but this seems like the sort of API where simpler really is better.

I'll try to take a pass over this variation and see if there's a not-so-bad way to have the best of both

wojtek-t committed:
Fix race in informer transformers

riendeau committed:
Bump base images for agnhost

Commit:
test: typecheck cel test cases

serathius committed:
Cleanup defer from SetFeatureGateDuringTest function call

serathius committed:
Upgrade etcd libraries to v3.5.13

Add otelgrpc.WithMessageEvents(otelgrpc.ReceivedEvents, otelgrpc.SentEvents) to tracing options due to https://github.com/open-telemetry/opentelemetry-go-contrib/pull/3964

serathius committed:
Remove workarounds for https://github.com/etcd-io/etcd/issues/17507

k8s-ci-robot committed:
Merge pull request #122657 from p0lyn0mial/upstream-client-go-rest-client-watch-list

add watchlist to client-go rest client

k8s-ci-robot committed:
Merge pull request #124027 from liangyuanpeng/etcd_update_3.5.13

etcd: update to v3.5.13

pohly committed:
hack: use "latest" apidiff

Pinning apidiff to a specific version shouldn't be necessary because, if past
experience in klog holds true, the latest version just works. This way we don't
have to remember to bump up the revision.

The downside of using "latest" is that a compromise of that version would give
an attacker the ability to run code in the CI and on developer machines.

neolit123 committed:
kubeadm: fix bug of kubeletconfig not being downloaded on upgrade

During upgrade apply we had logic to download the kubelet and
kubeproxy configs from the cluster as part of the call to:
  FetchInitConfigurationFromCluster()

With the introduction of UpgradeConfiguration there was
some refactor in this area and the function no longer
had the argument skipComponentConfigs set to false.

It is set to 'true', an InitConfiguration is downloaded
but it would contain empty / defaulted component configs.

- Set the argument to 'false'
- Perform minor cleanup of STDOUT messages and comments.
that were missed in 1.30.

ah8ad3 committed:
Add FeatureSupportChecker for etcd storage

k8s-ci-robot committed:
Merge pull request #124474 from serathius/allow-bookmarks

Allow bookmark events in between delete/modify in testSimpleCRUD

aojea committed:
use latest stable version of kube-network-policies for CI

wangzhen127 committed:
Add wangzhen127 back to approvers and reviewers for NPD

k8s-ci-robot committed:
Merge pull request #124161 from aojea/lb_timeout

fix e2e loadbalancer test timeouts and assumptions

k8s-ci-robot committed:
Merge pull request #124471 from aojea/zero_e2e

don't force delete pods on e2e tests

k8s-ci-robot committed:
Merge pull request #124484 from aojea/network_policies

use latest stable version of kube-network-policies for CI

k8s-ci-robot committed:
Merge pull request #124488 from wangzhen127/owners

Add wangzhen127 back to approvers and reviewers for NPD

aramase committed:
Revert "Run `controlplane/transformation` integration tests in parallel"

k8s-ci-robot committed:
Merge pull request #124075 from pohly/dra-api-comments

DRA: API cleanup + fix

k8s-ci-robot committed:
Merge pull request #124086 from matthyx/probes

add coverage tests for probes behavior

k8s-ci-robot committed:
Merge pull request #124492 from aramase/revert-124250-aramase/t/transformation_tests_parallel

Revert "Run `controlplane/transformation` integration tests in parallel"

brianpursley committed:
Improve performance of TestForceApply unit test.

Change backOffPeriod from const to var, so that it can be set lower during unit test.

k8s-ci-robot committed:
Merge pull request #123572 from alexzielenski/cel/name-formats

feature: add name formats library to CEL

k8s-ci-robot committed:
Merge pull request #124299 from kushalShukla-web/componentconfig

removed this 	k8s.io/kubernetes/pkg/apis/componentconfig from verify-…

carlory committed:
kubeadm: remove the deprecated output.kubeadm.k8s.io/v1alpha2

aroradaman committed:
agnhost: update README

dummy commit to trigger agnhost image promotion pipeline

Signed-off-by: Daman Arora <aroradaman@gmail.com>

carlory committed:
scheduler deprecates non-csi plugins

k8s-ci-robot committed:
Merge pull request #124468 from neolit123/1.31-improve-unit-test-performance-with-test-versions

kubeadm: improve performance of unit tests that need a k8s version

k8s-ci-robot committed:
Merge pull request #124480 from neolit123/1.31-fix-upgrade-component-config

kubeadm: fix bug of kubeletconfig not being downloaded on upgrade

jbtk committed:
Append both env variables instead of fetching twice and overriding them.
Previously the second line would fetch the clean environment variables
and drop the changes from the first line.

neolit123 committed:
kubeadm: check for available nodes during 'CreateJob' preflight

Instead of erroring during the preflight check 'CreateJob'
from "upgrade" commands when there are no schedulable nodes,
show a warning.

This can happen in single node clusters.

Also increase the Job TTL after completion to 20 seconds
to make sure it's more than the timeout what waits
for the Job to complete.

bells17 committed:
component-helpers: Support structured and contextual logging (#120637)

k8s-ci-robot committed:
Merge pull request #124498 from aroradaman/dummy-agnhost-bump

agnhost: update README

aojea committed:
Revert "added init to image-utils.sh"

This reverts commit 7951d7975e9d34b213b15fbc02debf02f5086f7e.

aojea committed:
Revert "Remove _tmp from test/images/image-util.sh"

This reverts commit 2e465cd6fa06165dfb50c0f1c39254203721dae7.

k8s-ci-robot committed:
Merge pull request #124263 from alvaroaleman/typed

Workqueue: Add generic versions that are properly typed

k8s-ci-robot committed:
Merge pull request #124469 from serathius/etcd-3.5.13

Upgrade etcd libraries to v3.5.13

k8s-ci-robot committed:
Merge pull request #122971 from pohly/apidiff

hack: add apidiff.sh

k8s-ci-robot committed:
Merge pull request #124452 from danwinship/proxy-config-unit-tests

KubeProxyConfig unit test cleanups

k8s-ci-robot committed:
Merge pull request #124508 from aojea/agnhost_fix

Agnhost building fix

aroradaman committed:
Revert "agnhost: update README"

bart0sh committed:
Revert "add coverage tests for probes behavior"

This reverts commit 9be9832184cb4069501c50dd8fdd001417e553af.

k8s-ci-robot committed:
Merge pull request #124510 from aroradaman/revert-124498-dummy-agnhost-bump

Revert "agnhost: update README"

danwinship committed:
Fix behavior with external cloud provider and --hostname-override

k8s-ci-robot committed:
Merge pull request #124344 from wojtek-t/fix_transformer

Fix race in informer transformers

k8s-ci-robot committed:
Merge pull request #124352 from wojtek-t/idempotent_transformer_func

Idempotent transformers in scheduler & kcm

k8s-ci-robot committed:
Merge pull request #123930 from Nordix/esotsal/fix_123928

Move DockerHung test in the end

k8s-ci-robot committed:
Merge pull request #124513 from serathius/workarounds

Remove workarounds for https://github.com/etcd-io/etcd/issues/17507

k8s-ci-robot committed:
Merge pull request #124466 from serathius/featuregate-defer

Cleanup defer from SetFeatureGateDuringTest function call

k8s-ci-robot committed:
Merge pull request #124512 from bart0sh/PR143-e2e-node-fix-containers-lifecycle

Revert "add coverage tests for probes behavior"

dims committed:
drop unused code pointed out by linter

Signed-off-by: Davanum Srinivas <davanum@gmail.com>

k8s-ci-robot committed:
Merge pull request #124382 from riendeau/agnhostbump

Bump base images for agnhost

k8s-ci-robot committed:
Merge pull request #124496 from carlory/output-alpha3

kubeadm: remove the deprecated output.kubeadm.k8s.io/v1alpha2

pohly committed:
hack/apidiff.sh: handle lack of origin

The script failed in the Prow job with:

   fatal: Not a valid object name origin/master

That came from "git merge-base origin/master HEAD", which isn't needed because
the Prow job intentionally sets -r explicitly.

Now the fallback is only tried if needed. While at it, the informational
messages get enhanced.

k8s-ci-robot committed:
Merge pull request #123777 from linxiulei/nc

Remove setting NoRouteCreated condition

pohly committed:
scheduler: move assume cache to utils, part 1

This is a verbatim move resp. copy of the files. They don't build in their new
location yet.

pohly committed:
scheduler: remove AssumeCache interface

There's no reason for having the interface because there is only one
implementation. Makes the implementation of the test functions a bit
simpler (no casting). They are still stand-alone functions instead of methods
because they should not be considered part of the "normal" API.

k8s-ci-robot committed:
Merge pull request #124521 from pohly/apidiff

hack/apidiff.sh: handle lack of origin

bart0sh committed:
node_e2e: refactor RunTogether function

jbtk committed:
Fix linter issues

irsl committed:
gitRepo volume: directory must be max 1 level deep

More details on Hackerone #2266560

aojea committed:
use a stable release for network policies

k8s-ci-robot committed:
Merge pull request #122145 from pohly/cloud-providers-context-support

cloud providers: enhance context support

aroradaman committed:
pkg/proxy/util/nfacct: utility to interact with nfacct subsystem

nfacct is netfilter's accounting subsystem. This utility allows
interactions with the subsystem using lower level netlink API.

Signed-off-by: Daman Arora <aroradaman@gmail.com>

k8s-ci-robot committed:
Merge pull request #124524 from jbtk/env-vars

Append both env variables instead of fetching twice and overriding them.

k8s-ci-robot committed:
Merge pull request #124516 from danwinship/cloud-hostname-override

Fix behavior with external cloud provider and --hostname-override

danwinship committed:
Rename nftables sync failure metric

danwinship committed:
Re-enable V(9) transaction logging in nftables proxy

SataQiu committed:
kubeadm: enable KubernetesReleaseVersion test

k8s-ci-robot committed:
Merge pull request #124545 from SataQiu/fix-kubeadm-test-20240426

kubeadm: enable KubernetesReleaseVersion test

k8s-ci-robot committed:
Merge pull request #124419 from neolit123/1.31-deprecate-experimental-cert-rotation

kubeadm: mark the phase 'experimental-cert-rotation' as deprecated

sttts committed:
kube-apiserver/aggregator: fix panic with crds disabled

Signed-off-by: Dr. Stefan Schimanski <stefan.schimanski@gmail.com>

pohly committed:
hack/apidiff.sh: compare between two revisions, usability enhancements

In a Prow job, the current work tree is the result of merging a PR into the
target. We want apidiff.sh from there, but then need to invoke it for two
specific revisions and compare.

While at it, output and usability get enhanced. The directory parameter(s) may
be absolute paths or lack the leading . that is required by apidiff.

sttts committed:
controlplane/apiserver: move peer proxy code to allow aggregator construction

This reverts commit 0bb6ff5e454cfc99c0b611e22162391eaa31e103.

Signed-off-by: Dr. Stefan Schimanski <stefan.schimanski@gmail.com>

k8s-ci-robot committed:
Merge pull request #124549 from sttts/sttts-aggregator-without-crds

kube-apiserver/aggregator: fix panic with crds disabled

sttts committed:
pkg/controlplane: split apart generic server part of instance.go

Signed-off-by: Dr. Stefan Schimanski <stefan.schimanski@gmail.com>

danwinship committed:
Add nftables cleanup failure metric, fix cleanup bug

If the sync fails, don't try to cleanup, since it's guaranteed to fail
too.

k8s-ci-robot committed:
Merge pull request #124552 from pohly/apidiff

hack/apidiff.sh: compare between two revisions, usability enhancements

riendeau committed:
Bump version of agnhost used in e2e tests

sbueringer committed:
Bump sigs.k8s.io/yaml to v1.4.0

chengjoey committed:
add e2e test when a plugin (in-tree or out-of-tree) return non-existent/illegal nodes, the pod should be scheduled normally if suitable nodes exists

Signed-off-by: joey <zchengjoey@gmail.com>

sttts committed:
Address comments

Signed-off-by: Dr. Stefan Schimanski <stefan.schimanski@gmail.com>

k8s-ci-robot committed:
Merge pull request #124102 from pohly/dra-scheduler-assume-cache

scheduler: move assume cache to utils

k8s-ci-robot committed:
Merge pull request #124503 from neolit123/1.31-convert-create-job-preflight-to-warning

kubeadm: check for available nodes during 'CreateJob' preflight

aojea committed:
tag TrafficDistribution tests to avoid running on clusters without the feature enabled

aojea committed:
tag service.spec.TrafficDistribution field as alpha

k8s-ci-robot committed:
Merge pull request #124494 from brianpursley/FixSlowUnitTest-TestForceApply

Improve performance of TestForceApply unit test (k8s.io/kubectl/pkg/cmd/apply)

sttts committed:
kube-apiserver/aggregator: remove unused PeerCAFile in config

Signed-off-by: Dr. Stefan Schimanski <stefan.schimanski@gmail.com>

enj committed:
Add aramase to sig-auth-authenticators-reviewers

k8s-ci-robot committed:
Merge pull request #124529 from danwinship/localtraffic-cleanup-1

LocalTrafficDetector implementation cleanup

k8s-ci-robot committed:
Merge pull request #124551 from sttts/sttts-kube-apiserver-config-split

pkg/controlplane: split up config into generic controlplane and kube-related part

k8s-ci-robot committed:
Merge pull request #124575 from sttts/sttts-unused-peerca

kube-apiserver/aggregator: remove unused PeerCAFile in config

Huang-Wei committed:
test to catch illegal nodeNames returned by PreFilterResult

k8s-ci-robot committed:
Merge pull request #123795 from saschagrunert/metrics-remote-runtime

Decouple `kubelet/cri/remote` package from `kubelet/metrics`

k8s-ci-robot committed:
Merge pull request #124557 from danwinship/metrics-and-stuff

kube-proxy metrics cleanup (and stuff)

k8s-ci-robot committed:
Merge pull request #124539 from Huang-Wei/ut-for-123465

test to catch illegal nodeNames returned by PreFilterResult

aroradaman committed:
pkg/proxy/iptables: track ct state invalid dropped packets

Track packets dropped by proxy which were marked invalid by conntrack
using nfacct netfilter extended accounting infrastructure.

Signed-off-by: Daman Arora <aroradaman@gmail.com>

sttts committed:
controlplane: split out SystemNamespaces

Signed-off-by: Dr. Stefan Schimanski <stefan.schimanski@gmail.com>

sanposhiho committed:
support `scheduler_plugin_execution_duration_seconds` in scheduler_perf

k8s-ci-robot committed:
Merge pull request #122812 from aroradaman/ct-invalid-drop-metric

Metric to track conntrack state invalid packets dropped by iptables

k8s-ci-robot committed:
Merge pull request #124560 from riendeau/agnhost-bump

Bump version of agnhost used in e2e tests

k8s-ci-robot committed:
Merge pull request #124577 from sttts/sttts-split-out-system-namespaces

prefactoring: generic controlplane: split out SystemNamespaces

aroradaman committed:
e2e/framework network: exit early in DialFromContainer

Immediately exit on receiving an unexpected response, we currently
wait for all the attempts to be exhausted and then return an error.

Signed-off-by: Daman Arora <aroradaman@gmail.com>

k8s-ci-robot committed:
Merge pull request #124583 from aroradaman/e2e-net-dial-from-container

e2e/framework network: exit early in DialFromContainer

k8s-ci-robot committed:
Merge pull request #124567 from chengjoey/e2e/scheduler-prefilter

add e2e test when a plugin (in-tree or out-of-tree) return non-existent/illegal nodes, the pod should be scheduled normally if suitable nodes exists

chengjoey committed:
add integration test for pod with pvc has node-affinity to non-existent/existent nodes

Signed-off-by: joey <zchengjoey@gmail.com>

carlory committed:
bugfix: resourceclaim forgot to wait for podSchedulingSynced and templatesSynced

xrstf committed:
fix stray +optional comment on persistentVolumeClaimRetentionPolicy

xrstf committed:
codegen

k8s-ci-robot committed:
Merge pull request #123984 from carlory/volume-remove-cloud-provider

Remove cloud provider dependency from volume host and volume controllers

ah8ad3 committed:
Add completion for kubectl set image

aroradaman committed:
agnhost/netexec: log hostname

Signed-off-by: Daman Arora <aroradaman@gmail.com>

carlory committed:
use slices library instead

aroradaman committed:
agnhost: version bump to v2.52

Signed-off-by: Daman Arora <aroradaman@gmail.com>

k8s-ci-robot committed:
Merge pull request #124147 from devincd/put-qps-cmd

Place the qps and burst parameters on the command line in kubemark

k8s-ci-robot committed:
Merge pull request #124594 from aroradaman/agnhost-log-requests

agnhost: log response for /hostname

TessaIO committed:
fix: add whitespace for drain error message

Signed-off-by: ahmed.g <ahmed.g@adjoe.io>

danwinship committed:
clean up LocalTrafficDetector construction / tests (#124582)

* LocalTrafficDetector construction and test improvements

* Reorder getLocalDetector unit test fields so "input" args come before "output" args

* Don't pass DetectLocalMode as a separate arg to getLocalDetector

It's already part of `config`

* Clarify test names in preparation for merging

* Merge single-stack/dual-stack LocalTrafficDetector construction

Also, only warn if the *primary* IP family is not correctly configured
(since we don't actually know if the cluster is really dual-stack or
not), and pass the pair of detectors to the proxiers as a map rather
than an array.

* Remove the rest of Test_getDualStackLocalDetectorTuple

alvaroaleman committed:
Use the generic/typed workqueue throughout

This change makes us use the generic workqueue throughout the project in
order to improve type safety and readability of the code.

aroradaman committed:
promote agnhost to v2.52

Signed-off-by: Daman Arora <aroradaman@gmail.com>

k8s-ci-robot committed:
Merge pull request #124599 from aroradaman/agnhost-2.52

promote agnhost to v2.52

xrstf committed:
alpha->beta

k8s-ci-robot committed:
Merge pull request #124576 from sttts/sttts-peer-proxy-generic-move

controlplane/apiserver: move peer proxy code to allow generic aggregator construction

k8s-ci-robot committed:
Merge pull request #120202 from sttts/sttts-controlplane-config-split

Step 2 – generic controlplane: split server

k8s-ci-robot committed:
Merge pull request #124483 from ah8ad3/etcd_feature_support

Add FeatureSupportChecker for etcd storage

k8s-ci-robot committed:
Merge pull request #123463 from k82cn/cri_grpc_rs

grpc: set localhost Authority to unix client calls

k8s-ci-robot committed:
Merge pull request #123796 from saschagrunert/kube-features-remote-runtime

Decouple `kubelet/cri/remote` package from `pkg/features`

k8s-ci-robot committed:
Merge pull request #124532 from aojea/netpol2.0

use a stable release for network policies

wojtek-t committed:
Deflake watchcache tests

k8s-ci-robot committed:
Merge pull request #123917 from DamianSawicki/dns-1-23-0

Update kube-dns and nodelocaldns to 1.23.0

sttts committed:
aggregator: split API priority map into generic and non-generic

Signed-off-by: Dr. Stefan Schimanski <stefan.schimanski@gmail.com>

sttts committed:
controlplane: return client error

Signed-off-by: Dr. Stefan Schimanski <stefan.schimanski@gmail.com>

sttts committed:
controlplane: cosmetics

Signed-off-by: Dr. Stefan Schimanski <stefan.schimanski@gmail.com>

k8s-ci-robot committed:
Merge pull request #121256 from liyuerich/workqueue

drop deprecated workqueue NewNamed package

k8s-ci-robot committed:
Merge pull request #124601 from enj/patch-9

Add aramase to sig-auth-authenticators-reviewers

aojea committed:
use log level 4 by default for the controller managers

k8s-ci-robot committed:
Merge pull request #124616 from aojea/ccm_logging

kube-up.sh: use log level 4 by default for the controller managers

aramase committed:
Remove unused NewSecure function in requestheader.go

Signed-off-by: Anish Ramasekar <anish.ramasekar@gmail.com>

sttts committed:
Address comments

Signed-off-by: Dr. Stefan Schimanski <stefan.schimanski@gmail.com>

k8s-ci-robot committed:
Merge pull request #122026 from skitt/sets-go1.21

Use Go 1.21 Ordered and clear for sets

k8s-ci-robot committed:
Merge pull request #124562 from sbueringer/pr-bump-sigs-yaml

Bump sigs.k8s.io/yaml to v1.4.0

sttts committed:
kube-apiserver: split admission initializers into generic and non-generic

Signed-off-by: Dr. Stefan Schimanski <stefan.schimanski@gmail.com>

sttts committed:
kube-apiserver: split up config creation into generic and non-generic part

Signed-off-by: Dr. Stefan Schimanski <stefan.schimanski@gmail.com>

k8s-ci-robot committed:
Merge pull request #123398 from ffromani/remove-legacy-checkpoint

node: devicemgr: remove obsolete pre-1.20 checkpoint file support

k8s-ci-robot committed:
Merge pull request #124572 from aojea/alpha_tag

tag TrafficDistribution tests to avoid running on clusters without th…

k8s-ci-robot committed:
Merge pull request #124613 from sttts/sttts-generic-api-versions

generic controplane: split aggregator API priorities

k8s-ci-robot committed:
Merge pull request #122148 from pohly/controllers-context-support

controllers + apiserver: enhance context support

k8s-ci-robot committed:
Merge pull request #124620 from sttts/sttts-controlplane-cosmetics

controlplane: cosmetics

k8s-ci-robot committed:
Merge pull request #124593 from carlory/dra-scheduler

use slices library instead of custom function

k8s-ci-robot committed:
Merge pull request #124610 from wojtek-t/deflake_cacher_tests

Deflake watchcache tests

neolit123 committed:
kubeadm: add support for custom cert validity period in v1beta4

Allow the user to pass custom cert validity period with
ClusterConfiguration.CertificateValidityPeriod and
CACertificateValidityPeriod.

The defaults remain 1 year for regular cert and 10 years for CA.
Show warnings if the provided values are more than the defaults.

Additional changes:
- In "certs show-expiration" use HumanDuration() to print
more detailed durations instead of ShortHumanDuration().
- Add a new kubeadm util GetStartTime() which can be used
to consistently get a UTC time for tasks like writing certs
and unit tests.
- Update unit tests to validate the new customizable NotAfter.

k8s-ci-robot committed:
Merge pull request #123396 from cyclinder/apiserver_datarace

Fix data race in apiserver mux handler

k8s-ci-robot committed:
Merge pull request #124597 from TessaIO/fix-drain-error-message

fix: add whitespace to drain error message

aroradaman committed:
e2e/network: stream pod logs for debugging

Intermittent logging to understand root cause of
https://github.com/kubernetes/kubernetes/issues/123760

Signed-off-by: Daman Arora <aroradaman@gmail.com>

matthyx committed:
e2e lifecycle: use millisecond resolution for logs

Signed-off-by: Matthias Bertschy <matthias.bertschy@gmail.com>

k8s-ci-robot committed:
Merge pull request #124450 from muff1nman/handle-nil-k8s-client

apiserver/options: avoid segfault by handling unset core k8s client

k8s-ci-robot committed:
Merge pull request #123128 from yue9944882/admission-runtime-class-empty-map-fix

Fail admission check upon nil/empty overhead map

k8s-ci-robot committed:
Merge pull request #124287 from sanposhiho/tainttoleration

implement QueueingHint in TaintToleration

k8s-ci-robot committed:
Merge pull request #124621 from sttts/sttts-controlplane-admission-initializers

kube-apiserver: split admission initializers into generic and non-generic

k8s-ci-robot committed:
Merge pull request #124644 from aroradaman/temp-pod-logs

e2e/network: stream pod logs for debugging

aroradaman committed:
kubectl/describe/svc: refactor TestDescribeService

Signed-off-by: Daman Arora <aroradaman@gmail.com>

aroradaman committed:
kubectl/describe: use endpointslices for describing endpoints

Signed-off-by: Daman Arora <aroradaman@gmail.com>

dims committed:
address comments during review

Signed-off-by: Davanum Srinivas <davanum@gmail.com>

sttts committed:
aggregator: move construction to generic controlplane

Signed-off-by: Dr. Stefan Schimanski <stefan.schimanski@gmail.com>

sttts committed:
Make linter happy

Signed-off-by: Dr. Stefan Schimanski <stefan.schimanski@gmail.com>

k8s-ci-robot committed:
Merge pull request #123756 from saschagrunert/cri-approvers

Sync `cri-api` approvers with kubelet `cri` package

k8s-ci-robot committed:
Merge pull request #124080 from claudiubelu/skip-windows-tests

Skip failing Windows tests

k8s-ci-robot committed:
Merge pull request #124463 from carlory/remove-fg-ConsistentHTTPGetHandlers

remove feature-gate ConsistentHTTPGetHandlers

k8s-ci-robot committed:
Merge pull request #124638 from sttts/sttts-kube-apiserver-config-split-up

kube-apiserver: split up config creation into generic and non-generic part

matthyx committed:
e2e lifecycle: fix finishing -> exiting

Signed-off-by: Matthias Bertschy <matthias.bertschy@gmail.com>

k8s-ci-robot committed:
Merge pull request #124619 from aramase/aramase/c/requestheader_cleanup

Remove unused NewSecure function in requestheader.go

sttts committed:
controlplane/apiserver: unify options package name

Signed-off-by: Dr. Stefan Schimanski <stefan.schimanski@gmail.com>

k8s-ci-robot committed:
Merge pull request #124654 from sttts/sttts-controlplane-aggregator

Step 9 – aggregator: move construction to generic controlplane

matthyx committed:
e2e lifecycle: increase delay for restartable init containers

Signed-off-by: Matthias Bertschy <matthias.bertschy@gmail.com>

k8s-ci-robot committed:
Merge pull request #124645 from matthyx/ms

e2e lifecycle: reduce test flackiness

jpbetz committed:
Promote RetryGenerateName to beta

saschagrunert committed:
Fix `cri-api` `SECURITY_CONTACTS` by removing a duplicate

`tallclair` is already part of the list, so we can remove that entry.

Signed-off-by: Sascha Grunert <sgrunert@redhat.com>

neolit123 committed:
kubeadm: update the IsPriviligedUser preflight check on Windows

Use GetCurrentProcessToken() instead of checking the groups of a user.

The Go stdlib way of fetching the groups of an user appears
to be failing on some Windows setups. Which could be a regression
in later Go versions, or simply the code does not work on certain
setups.

gabesaba committed:
Add unit test which checks Gated is set/unset properly

k8s-ci-robot committed:
Merge pull request #123797 from saschagrunert/cri-staging

Add `cri-client` staging repository

k8s-ci-robot committed:
Merge pull request #124669 from gabesaba/test_gated

Add unit test which checks QueuedPodInfo.Gated matches its source

k8s-ci-robot committed:
Merge pull request #124663 from saschagrunert/cri-api-security-contacts

Fix `cri-api` `SECURITY_CONTACTS` by removing a duplicate

k8s-ci-robot committed:
Merge pull request #124665 from neolit123/1.31-fix-windows-priv-preflight-check

kubeadm: update the IsPriviligedUser preflight check on Windows

seantywork committed:
Updated & added visibility to apiserver x509 test certificates expiring this year

k8s-ci-robot committed:
Merge pull request #124662 from seantywork/master

Updated & added visibility to apiserver x509 test certificates expiring this year

SataQiu committed:
kubeadm: remove unused CreateKubeConfigFileFunc type

k8s-ci-robot committed:
Merge pull request #124684 from SataQiu/clean-20240503

kubeadm: remove unused CreateKubeConfigFileFunc type

k8s-ci-robot committed:
Merge pull request #124666 from chengjoey/ut-for-123465

add integration test for pod with pvc has node-affinity to non-existent/illegal nodes

dims committed:
Fixes from review by Jan Šafránek/Andrew/Michael

Signed-off-by: Davanum Srinivas <davanum@gmail.com>

k8s-ci-robot committed:
Merge pull request #121496 from benluddy/metav1-labelselector-fuzz

Deduplicate set expression values in metav1.LabelSelector fuzzer.

pmalek committed:
fix(api): make LocalObjectReference.Name and HostAlias.IP required (#124553)

* fix(api): LocalObjectReference Name a "" default and make HostAlias.IP required

* chore(api): add LocalObjectReference comment

* chore(api): add omitempty to LocalObjectReference's Name

* chore(api): add kubebuilder:default annotation

* chore(api): ./hack/update-codegen.sh

k8s-ci-robot committed:
Merge pull request #124531 from irsl/master

gitRepo volume: directory must be max 1 level deep

k8s-ci-robot committed:
Merge pull request #124189 from toddtreece/kube-aggregator-proxy-tracing

Add tracing to kube-aggregator proxyHandler

k8s-ci-robot committed:
Merge pull request #123886 from adrianreber/2024-03-12-criu-not-found

Handle containerd "CRIU not found" error message

k8s-ci-robot committed:
Merge pull request #124281 from fulviodenza/master

Allow request parameter and proto file path

mowangdk committed:
chore:update invalid package name by golint

k8s-ci-robot committed:
Merge pull request #124578 from sanposhiho/scheduler_perf_scheduler_plugin_execution_duration_seconds

support `scheduler_plugin_execution_duration_seconds` in scheduler_perf

k8s-ci-robot committed:
Merge pull request #124131 from zyjhtangtang/master

Correct variable names in EndpointSliceMirroring controller

k8s-ci-robot committed:
Merge pull request #124693 from mowangdk/feat/update_invalid_golint_package_name

chore: fix invalid package name by golint

k8s-ci-robot committed:
Merge pull request #119781 from huweihuang/fix/fix-link-failure

fix link failure for changelog

k8s-ci-robot committed:
Merge pull request #124589 from carlory/dra-resourceclaim-controller

the resourceclaim controller forgot to wait for podSchedulingSynced and templatesSynced

k8s-ci-robot committed:
Merge pull request #124590 from xrstf/fix-optional-comment

Fix +optional in comment for persistentVolumeClaimRetentionPolicy

k8s-ci-robot committed:
Merge pull request #124500 from carlory/scheduler-deprecate-non-csi-plugins

scheduler deprecates non-csi volumelimit plugins

k8s-ci-robot committed:
Merge pull request #124519 from dims/drop-all-the-providery-things-take-2

Remove gcp in-tree cloud provider and credential providers

jpbetz committed:
Promote CRD field selectors to beta

k8s-ci-robot committed:
Merge pull request #124598 from aroradaman/kubectl-describe-svc

Use endpointSlices for describing service endpoints

k8s-ci-robot committed:
Merge pull request #124668 from bart0sh/PR143-e2e-node-fix-containers-lifecycle

node_e2e: refactor RunTogether function

k8s-ci-robot committed:
Merge pull request #124673 from jpbetz/retry-beta

Promote RetryGenerateName to beta

jpbetz committed:
Add version testing

k8s-ci-robot committed:
Merge pull request #124600 from alvaroaleman/typed-wq

Use the generic/typed workqueue throughout

sanposhiho committed:
scheduler: preallocation for NodeToStatusMap

k8s-ci-robot committed:
Merge pull request #124592 from ah8ad3/add-kubectl-set-completion

Add completion for kubectl set image

bells17 committed:
Include k8s.io components with contextual logging in logcheck.conf

k8s-ci-robot committed:
Merge pull request #124722 from bells17/update-logcheck-k8s.io-components

Include k8s.io components with contextual logging in logcheck.conf

enj committed:
Sync go.work.sum of kms mock plugin

Signed-off-by: Monis Khan <mok@microsoft.com>

k8s-ci-robot committed:
Merge pull request #124714 from sanposhiho/prealloc

scheduler: preallocation for NodeToStatusMap

k8s-ci-robot committed:
Merge pull request #124726 from enj/enj/i/kms_mock_bump_work

Sync `go.work.sum` of kms mock plugin

k8s-ci-robot committed:
Merge pull request #124652 from neolit123/1.31-add-cert-expiration-v1beta4

kubeadm: add support for custom cert validity period in v1beta4

k8s-ci-robot committed:
Merge pull request #124681 from jpbetz/field-selector-beta

Promote custom resource field selectors to beta

thockin committed:
HACK: annotate gate-setting flags for later searching

thockin committed:
Add a new feature-gate library

thockin committed:
Deomonstrate GateSet composition

This commit demonstrate how libraries can define and expose Gates and
GateSets, and compose GateSets into new GateSets.  It also defines a small
number of new-style gates, but does not use them yet.

thockin committed:
Plumb new gates logic into binaries

TODO: incomplete

thockin committed:
FIXME: break this up

thockin committed:
WIP: flag plumbing

Notes:

Some of these parse into a map in their config and then call SetFromMap -- need to retain ability to pass in via config.
Option 1: make a `feature.EnableMapControl()`
Option 2: pass map into `feature.EnablePFlagControl()`
Option 3: `feature.GetAllGates(feature.FromFlags(flagName), feature.FromEnv(prefix))`
	This is probablay most correct today and makes validation easy,
	since we do it exactly once.  A little awkward with any dynamic
	source (e.g. API).

Should env enablement be part of "AddFlags" logic in these components?
It's a little muddy how apps are configured.

```
$ for f in _output/bin/*; do X=$($f --help 2>&1 | grep gates); if [[ -z "$X" ]]; then X=$($f init --help 2>&1 | grep gates); fi; if [[ -n "$X" ]]; then echo; echo "$f"; echo "$X"; fi; done

_output/bin/apiextensions-apiserver
      --feature-gates mapStringBool                             4444 A set of key=value pairs that describe feature gates for alpha/experimental features. Options are:
      --new-feature-gates mapStringBool                         A set of key=value pairs that describe feature gates for alpha/experimental features. Options are:

_output/bin/e2e_node.test
      --feature-gates mapStringBool                                6666 A set of key=value pairs that describe feature gates for alpha/experimental features.
      --service-feature-gates mapStringBool                        7777 A set of key=value pairs that describe feature gates for alpha/experimental features for API service.

_output/bin/kubeadm
      --feature-gates string                 2222 A set of key=value pairs that describe feature gates for various features. Options are:
      --new-feature-gates mapStringBool      A set of key=value pairs that describe feature gates for alpha/experimental features. Options are:

_output/bin/kube-aggregator
      --feature-gates mapStringBool                             4444 A set of key=value pairs that describe feature gates for alpha/experimental features. Options are:
      --new-feature-gates mapStringBool                         A set of key=value pairs that describe feature gates for alpha/experimental features. Options are:

_output/bin/kube-apiserver
      --feature-gates mapStringBool                        4444 A set of key=value pairs that describe feature gates for alpha/experimental features. Options are:
      --new-feature-gates mapStringBool                    A set of key=value pairs that describe feature gates for alpha/experimental features. Options are:

_output/bin/kube-controller-manager
      --feature-gates mapStringBool              4444 A set of key=value pairs that describe feature gates for alpha/experimental features. Options are:
      --new-feature-gates mapStringBool          A set of key=value pairs that describe feature gates for alpha/experimental features. Options are:

_output/bin/kubelet
      --feature-gates mapStringBool                              3333 A set of key=value pairs that describe feature gates for alpha/experimental features. Options are:
      --new-feature-gates mapStringBool                          A set of key=value pairs that describe feature gates for alpha/experimental features. Options are:

_output/bin/kube-proxy
      --feature-gates mapStringBool                  1111 A set of key=value pairs that describe feature gates for alpha/experimental features. Options are:
      --new-feature-gates mapStringBool              A set of key=value pairs that describe feature gates for alpha/experimental features. Options are:

_output/bin/kube-scheduler
      --feature-gates mapStringBool   4444 A set of key=value pairs that describe feature gates for alpha/experimental features. Options are:
      --new-feature-gates mapStringBool   A set of key=value pairs that describe feature gates for alpha/experimental features. Options are:

```
